eksctl/eksctl-cluster-config.yaml

```diff
@@ -0,0 +1,252 @@
+#   upgrade:  eksctl upgrade cluster --config-file=eksctl-cluster-config.yaml
+#   delete:   eksctl delete cluster --config-file=eksctl-cluster-config.yaml
+#
+# Node group operations:
+# ref: https://eksctl.io/usage/managing-nodegroups/
+#
+#   eksctl create nodegroup --config-file=eksctl-cluster-config.yaml --include user-b,worker-b
+#   eksctl delete nodegroup --cluster jmte --name user-a
+#
+# Attribution: this was based on @yuvipanda's work in 2i2c! <3
+# ref: https://github.com/2i2c-org/pangeo-hubs/blob/8e552bc198d8339efe8c003cb847849255e8f8ed/aws/eksctl-config.yaml
+#
+
+
+
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+metadata:
+  name: jmte
+  # region:
+  #   The region was chosen to to us-west-2 (Oregon) to be close to a CMIP-6
+  #   dataset.
+  #
+  region: us-west-2
+  version: "1.19"
+  tags:
+    2i2c.org/project: jmte
+
+# availabilityZones:
+#   For the EKS control plane, arbitrary chosen but made explicit to ensure we
+#   can locate the node pool on an AZ where the EKS control plane exist as
+#   required.
+#
+availabilityZones: [us-west-2d, us-west-2b, us-west-2a]
+
+
+
+# This section will create additional k8s ServiceAccount's that are coupled with
+# AWS Role's. By declaring pods to use them, you can grant these pods the
+# associated permissions. For this deployment, we create a k8s ServiceAccount
+# with Full S3 credentials which we then also declare user pods and dask worker
+# pods will make use of.
+#
+iam:
+  withOIDC: true        # https://eksctl.io/usage/security/#withoidc
+  # serviceAccounts like nodeGroups etc can be managed directly with eksctl, for
+  # more information, see: https://eksctl.io/usage/iamserviceaccounts/
+  #
+  #   eksctl create iamserviceaccount --config-file=eksctl-cluster-config.yaml
+  #
+  # serviceAccounts:
+  #   # The 2i2c Helm chart creates a k8s ServiceAccount referenced as user-sa.
+  #   # I'm not confident on what happens by declaring this again here.
+  #   - metadata:
+  #       name: user-sa
+  #       namespace: default
+  #       labels:
+  #         aws-usage: application
+  #     attachPolicyARNs:
+  #       - arn:aws:iam::aws:policy/AmazonS3FullAccess
+
+
+
+# Choose the type of node group?
+# - nodeGroups cannot be updated but must be recreated on changes:
+#   https://eksctl.io/usage/managing-nodegroups/#nodegroup-immutability
+# - managedNodeGroups cannot scale to zero:
+#   https://github.com/aws/containers-roadmap/issues/724
+#
+# Choosing instance type?
+# - Maximum pods: https://github.com/awslabs/amazon-eks-ami/blob/master/files/eni-max-pods.txt
+# - Node specs:   https://aws.amazon.com/ec2/instance-types/
+# - Cost:         https://ec2pricing.net/
+#
+# Management advice:
+# - Always use a suffix for node group names that you can replace with something
+#   to create a new node group and delete the old. You will run into issues if
+#   you name it "core" and "core-a" instead of "core-a" and "core-b", such as
```

Comment:
+1, good idea. This has been another advantage with kops though, you can update node-pools in place.